EIPS/eip-6145.md

```diff
@@ -0,0 +1,84 @@
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+## Rationale
+
+A standardized way to deposit, withdraw and retrieve creator fee (“royalty”) information associated with a human readable name - this standard enables universal support for royalty payments across social networks, NFT marketplaces and ecosystem participants, its generic design can be implemented on any EVM compatible blockchain and its specific implementation is intended to associate royalties with a hashtag (#) name, however, it is highly versatile and can be used to attribute royalties for any Ethereum address, for example, an NFT.
+
+## Backwards Compatibility
+
+The standard contracts in implementation may be made upgradable to help ensure backward compatibliity
+
+### Overview
+
+| # | Smart Contract | Description |
+|---|----------------|-------------|
+| 1 | HashtagNFTCollectiveTreasury | The royalty treasury holds the creator fees payable in the EVM’s blockchain native token associated with a particular set of Ethereum addresses. |
+| 2 | HashtagNFTCollectiveResolver | Provides naming resolution functionality to retrieve Ethereum addresses from human readable names (e.g. “hashtags”). |
+
+### HashtagNFTCollectiveTreasury
+
+| # | Function | Input / Output | Description |
+|---|----------|----------------|-------------|
+| 1 | DepositRoyalty _(payable)_ | **Inputs** _address[] calldata tokenAddresses_ - Array of Ethereum addresses _uint256[] calldata tokenRoyalties_ - Matching array of royalty deposit amounts for each Ethereum Address  **Outputs** none  emits **Deposit** event _event Deposit(senderAddr, address[] tokenAddresses, uint256[] tokenRoyalties);_ | This method optimises the gas costs relating to the deposit of royalties for a number of different associated Ethereum addresses. It takes an array of tokenAddresses (Ethereum addresses) associated with an array of tokenRoyalties (Royalty amounts).  tokenAddresses array **MUST** be paired equally and **MUST** be in the same order as tokenRoyalties. The deposit payable amount sent to this function **MUST** be in the blockchain’s native token (for example: ETH, GLMR, MOVR, CELO, UNQ etc). The payable amount of the native token **MUST** equal the sum of all tokenRoyalties. A human readable name registered in the naming service **MAY** be associated with the Ethereum address(es) provided in tokenAddresses. The implementation of this method stores the sender address in a “whitelist” along with the quantity of the blockchain’s native token deposited which is in turn is used to permission the method WithdrawFromTreasuryToAddress. |
```

Comment:
I see a couple "MUST"s in here. If you're setting out requirements, they have to go in the Specification section.

The Backwards Compatibility section should just be for describing breaking changes. For example, if you were extending EIP-721 NFTs into soulbound tokens, you'd note here that soulbound tokens can't be transferred (unlike regular 721 NFTs).